src/types.ts

```diff
@@ -1722,3 +1722,146 @@ export type {
  * @public
  */
 export const EXPERIMENTAL_API_WARNING = 'This is an experimental API version'
+
+// Media Libraries types
+/**
+ * Fit / resize modes accepted for thumbnail params.
+ * @public
+ */
+export type FitMode = 'preserve' | 'stretch' | 'crop' | 'smartcrop' | 'pad'
+
+/**
+ * Allowed still image formats (thumbnail + storyboard).
+ * @public
+ */
+export type StillImageFormat = 'jpg' | 'png' | 'webp'
+
+/**
+ * Allowed animated image formats.
+ * @public
+ */
+export type AnimatedImageFormat = 'gif' | 'webp'
+
+/**
+ * Thumbnail rendition (single frame) options.
+ * @public
+ */
+export interface ThumbnailTransformOptions {
+  /** Pixel width of the thumbnail frame. */
+  width?: number
+  /** Pixel height of the thumbnail frame. */
+  height?: number
+  /** Timestamp (seconds) from which to grab the frame. */
+  time?: number
+  /** Resize / fit mode applied to the extracted frame. */
+  fit?: FitMode
+  /** Output image format. */
+  format?: StillImageFormat
+}
+
+/**
+ * Animated preview rendition options (e.g. GIF / animated WebP).
+ * @public
+ */
+export interface AnimatedTransformOptions {
+  /** Pixel width of the animated output. Max 640 px. */
+  width?: number
+  /** Pixel height of the animated output. Max 640 px. */
+  height?: number
+  /** Start time in seconds (inclusive). */
+  start?: number
+  /** End time in seconds. */
+  end?: number
+  /** Frames per second (1–30). */
+  fps?: number
+  /** Output animated format. */
+  format?: AnimatedImageFormat
+}
+
+/**
+ * Storyboard (contact sheet) options.
+ * @public
+ */
+export interface StoryboardTransformOptions {
+  /** Output image format for the storyboard. */
+  format?: StillImageFormat
+}
+
+/**
+ * Video-specific playback transformation option groups.
+ * Only explicitly provided values are serialized into query parameters.
+ * @public
+ */
+export interface MediaLibraryVideoPlaybackTransformations {
+  /** Static thumbnail (single frame) options. */
+  thumbnail?: ThumbnailTransformOptions
+  /** Animated preview options (GIF / animated WebP). */
+  animated?: AnimatedTransformOptions
+  /** Storyboard (contact sheet) options. */
+  storyboard?: StoryboardTransformOptions
+}
+
+/**
+ * Backwards compatibility alias.
+ * @deprecated Use MediaLibraryVideoPlaybackTransformations instead.
+ * @public
+ */
+export type MediaLibraryPlaybackTransformations = MediaLibraryVideoPlaybackTransformations
```

Comment:
What's the backwards compatibility for here? Is there existing usage of this type from somewhere else?